src/main/scala/sparkz/core/network/NetworkController.scala

```diff
@@ -206,7 +206,7 @@ class NetworkController(settings: NetworkSettings,
     log.info(s"Unconfirmed connection: ($remoteAddress, $localAddress) => $connectionId")
 
     val handlerRef = sender()
-    if (connectionDirection.isOutgoing && canEstablishNewOutgoingConnection) {
+    if (connectionDirection.isOutgoing && (canEstablishNewOutgoingConnection || canEstablishNewForgerConnection)) {
       createPeerConnectionHandler(connectionId, handlerRef)
     }
     else if (connectionDirection.isIncoming && canEstablishNewIncomingConnection)
```

Comment:
I intentionally left it like this, because otherwise, it would allow spamming a node - a potential security breach.

So when we just initialize the handshake, we only allow the outgoing connections to be over the limit - the ones that are initiated by the current node.
But if the other nodes would be allowed to go over the limit, they will spam us with connection attempts, and we will have to do a full handshake every time until we can decide that the node is a forger.